pkg/skaffold/errors/errors.go

```diff
@@ -36,46 +38,73 @@ const (
 var (
 	// ErrNoSuggestionFound error not found
 	ErrNoSuggestionFound = fmt.Errorf("no suggestions found")
+
+	setOptionsOnce sync.Once
+	skaffoldOpts   config.SkaffoldOptions
 )
 
 type Phase string
 
+// SetSkaffoldOptions set Skaffold config options once. These options are used later to
+// suggest actionable error messages based on skaffold run context
+func SetSkaffoldOptions(opts config.SkaffoldOptions) {
+	setOptionsOnce.Do(func() {
+		skaffoldOpts = opts
+	})
+}
+
 // ActionableErr returns an actionable error message with suggestions
 func ActionableErr(phase Phase, err error) *proto.ActionableErr {
-	aiErr := proto.ActionableErr{
-		ErrCode:     defaultErrorCodeFromError(phase),
+	errCode, suggestions := getErrorCodeFromError(phase, err)
+	return &proto.ActionableErr{
+		ErrCode:     errCode,
 		Message:     err.Error(),
-		Suggestions: nil,
+		Suggestions: suggestions,
 	}
-
-	return &aiErr
 }
 
-func ShowAIError(err error, opts config.SkaffoldOptions) error {
+func ShowAIError(err error) error {
 	for _, v := range knownBuildProblems {
 		if v.regexp.MatchString(err.Error()) {
```

Comment:
I think it would make sense to think of `problem` as a type and instead have a function `Matches(error)`.  This can be done later.

Reply:
sounds good.  I created an todo here #4389